### ISSUE TITLE
Feature request: Improve precision for the color toning UI

### ISSUE THREAD

**Issue by rebio** [Beep6581/RawTherapee#6362]
### Short description
The color toning UI currently is no 'precise mode' for selecting the colour (unlike the HSV Equalizer where you can hold down Ctrl to be more precise and Shift to snap it to eg. the minimum)

### Ideas in detail
The UI currently looks like this:
![default](https://user-images.githubusercontent.com/10911084/133638422-cde141da-2c1c-414b-944a-2e94fac82a96.png)

The suggestion is to add these functions:
- precise mode
- vector mode
- circle mode
- expand window button

**precise mode**
Used with: hold Ctrl + drag point
What it does: The movement of the mouse is linearly scaled down before it is applied to the point. The same can currently be done in the HSV Equalizer.
![01 precision mode](https://user-images.githubusercontent.com/10911084/133639313-6b3c16ce-ca25-433b-9048-1871534484e8.png)

**vector mode**
Used with: hold Shift + drag point
What it does: Moves the point only on the current vector that goes from the field origin and stretches through the current position of the point. When the Shift key is held down, the point will only move on that axis.
![02 vector mode](https://user-images.githubusercontent.com/10911084/133639698-851b376f-63dd-4b80-b378-95d8d5ddf801.png)

**circle mode**
Used with: hold Alt + drag point
What it does: The point will only be moved on a circle around the point. The circle radius is the current distance measured from the plane field origin to the current position of the point. When the Alt key is held down, the point will only move on this circle
![03 circle mode](https://user-images.githubusercontent.com/10911084/133639996-1d812c13-177d-4142-ac46-9bebef477166.png)

**expand window button**
Used with: Left click on the a UI button on the right side of the field
What it does: The size of the field is temporarily expanded. The field can be used just like the small one.
![04 expand window](https://user-images.githubusercontent.com/10911084/133641216-488a3c40-56ca-428e-b851-b01cfada9217.png)

UI with enlarged version:
![05 expand window-large](https://user-images.githubusercontent.com/10911084/133641226-423c62b8-4a1a-4dd0-8ce1-70d0b1efbb8a.png)

UI button for area enlargement (if there isn't already one).
Edit: The original one has a white box to fit into the default UI theme. The black box is for GitHub
![06 expand button](https://user-images.githubusercontent.com/10911084/133641311-7bb3889a-0e68-49f7-8d00-59fa8e72c3a4.png)![06a expand button](https://user-images.githubusercontent.com/10911084/133642824-286b73b3-3049-44e0-acf1-5b3bef3d4308.png)

### Usage
The use for this feature is to be able to be more precise when you select a colour. Currently it can sometimes be quite hard to nail the exact spot that you want to have. These changes would make them easier.

So that the operation does not need an explanation and is clear to everybody using it without any description, it could display small UI elements like in the mockups. However, as far as I have seen this would be a 'first' in RT so I understand it would take some time if it's feasible at all with the current setup.

**Comment by simohno** [comment]
I agree and congratulate with this idea, it is currently somewhat difficult to finely regulate the point position so to correct the color exactly as you would like.

**Comment by rebio** [comment]
An idea for the enlarged colour picker: I went through the preferences and tested out to have the editor in a different window and thereby use two displays (a good idea if you have a colour calibrated monitor, but it's not large enough to waste space on UI elements). Since RT knows how to open a video and can change the content in one Window based on changes done in a different one, the enlarge button could just open a new small (aka not full sized) Window with only the tool inside.

This function to enlarge the tool or _tool to new window_ would help with all curves and improve accuracy.

The thought is based on a current problem: I want to do slight changes to an image and want to be as precise as I can. I can enlarge the toolbar on the right to make eg. the curves modifiers bigger, but I I either waste a bunch of screen space (when I keep the UI size even I don't need it when I switch over to sliders) or I would have to resize the side bar over and over again.
